src/components/CreateAccountForm/CreateAccountForm.js

```diff
@@ -0,0 +1,323 @@
+      // Ignore zip code if user
+    } else if (this.state.address && this.state.city && this.state.USState) {
+      return this.state.address + ', ' + this.state.city + ', ' + this.state.USState;
+    } else {
+      return null;
+    }
+  }
+
+  verifyPassword(password, confirmPassword) {
+    if (password !== confirmPassword) return false; // passwords do not match
+    if (password.length < 6) return false; // password too short
+    if (password.toUpperCase() === password) return false; // no lowercase
+    if (password.toLowerCase() === password) return false; // no uppercase
+    if (!/\d/.test(password)) return false // no number
+
+    return true;
+  }
+
+  handleUserCreateSubmit = async (e) => {
+    e.preventDefault();
+
+    if (this.verifyPassword(this.state.password, this.state.confirmPassword)) {
+      await UsersService.createUser({
+        firstName: this.state.firstName,
+        lastName: this.state.lastName,
+        email: this.state.email,
+        password: this.state.password,
+        address: this.getFullAddress(),
+        zipCode: this.state.zipCode,
+      }).then(() => {
+        AuthService.loginUser({
+          email: this.state.email,
+          password: this.state.password
+        }).then((token) => {
+          AuthService.setToken(token);
+          this.context.setLoggedInState(true);
+        })
+      })
+        .catch(() => {
+          this.setState({ errorText: 'Account creation failed' })
+        });
+    } else {
+      this.setState({ errorText: 'Invalid password' });
+    }
+  };
+
+  handleShelterCreateSubmit = async (e) => {
+    e.preventDefault();
+
+    if (this.verifyPassword(this.state.password, this.state.confirmPassword)) {
+      await SheltersService.createShelter({
+        shelterName: this.state.shelterName,
+        emailAddress: this.state.email,
+        phoneNumber: this.state.phoneNumber,
+        password: this.state.password,
+        address: this.getFullAddress(),
+        website: this.state.website,
+      }).then(() => {
+        AuthService.loginShelter({
+          email: this.state.email,
+          password: this.state.password
+        }).then((token) => {
+          AuthService.setToken(token);
+          this.context.setLoggedInState(true);
+          this.context.setShelterAdminState(true);
+        })
+      })
+        .catch(() => {
+          this.setState({ errorText: 'Account creation failed' })
+        });
+    } else {
+      this.setState({ errorText: 'Invalid password' });
+    }
+  };
+
+  renderErrorMessage() {
+    if (this.state.errorText && this.state.errorText !== '') {
+      return (
+        <>
+          <div className='alert alert-danger'>{this.state.errorText}</div>
+        </>
+      )
+    }
+  }
+
+  renderUSStates() {
+    return (
+      <>
+        <option></option><option>AL</option><option>AK</option><option>AZ</option><option>AK</option><option>CA</option><option>CO</option><option>CT</option><option>DC</option>
```

Comment:
What do you think about dynamically generate this `<option>` by using loop or something (but not using `while` loop to avoid infinite loop)? Maybe you can store the states in a json object, and loop through it? I don't think we will change the option format in the future, but it will make it easier in the future if we want to.

Reply:
Updated!